lib/reply.js

```diff
@@ -591,6 +591,16 @@ function onSendEnd (reply, payload) {
     return
   }
 
+  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204) {
+    // Responses without a content body must not send content-type
+    // or content-length headers.
+    // See https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6.
+    reply.removeHeader('content-type')
```

Comment:
`content-length` should also be set to `0`

Reply:
>A server MUST NOT send a Content-Length header field in any response with a status code of [1xx (Informational)](https://www.rfc-editor.org/rfc/rfc9110.html#status.1xx) or [204 (No Content)](https://www.rfc-editor.org/rfc/rfc9110.html#status.204).

`content-length` should be `undefined` in `204` and `1xx` right ?

Reply:
`0` should be fine, that's what we have when the `payload` is explicitly `null` or `undefined`

Reply:
@Iamshankhadeep

Reply:
so we set the `content-length` as even if the status code is `204` or `1xx` ?

Reply:
Ah! Nvm, I misread this; my bad 👍